File: test/fixtures/sample.cpp
namespace demo {

enum Mode {
    Fast,
    Slow,
};

class Widget {
public:
    int value() const {
        return 1;
    }
};

template <typename T>
class Box {
public:
    T value() const {
        return value_;
    }

private:
    T value_;
};

template <typename T>
T identity(T value) {
    return value;
}

int build_widget() {
    Widget widget;
    return widget.value();
}

}

File: test/fixtures/sample.rs
pub struct Widget {
    value: i32,
}

pub struct Store<T> {
    value: T,
}

pub enum Mode {
    Fast,
    Slow,
}

impl Widget {
    pub fn new(value: i32) -> Self {
        Self { value }
    }
}

pub fn build_widget() -> Widget {
    Widget::new(1)
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn builds_widget() {
        let widget = build_widget();
        assert_eq!(widget.value, 1);
    }
}
